File: ClientApp/src/app/clases/alumnomateria.ts
export class alumnomateria{
    id : number;
     
    idalumno : number;
    idmateria : number;
    idciclolectivo : number;
    
    constructor(obj:any){
        //el orden de os parametros afecta la vista de los input en el modal
        this.id = +obj.id;
        this.idalumno = +obj.idalumno;
        this.idmateria =  +obj.idmateria;
        this.idciclolectivo = +obj.idciclolectivo;
       
         // formateo a number el string // parseInt(val

    }
    
}

File: modelo/alumnomateria.cs
using System.Data;
using System;
public class alumnomateria : oObjeto
{
    public int Idalumno { get; set; }
    public int Idmateria { get; set; }
    public int Idciclolectivo { get; set; }
    public alumnomateria() { }
    public alumnomateria(DataRow dr)
    { 
        this.Idalumno = Convert.ToInt32(dr["idalumno"]);
        this.Idmateria = Convert.ToInt32(dr["idmateria"]);
        this.Idciclolectivo = Convert.ToInt32(dr["idciclolectivo"]);
    }

}